### PR TITLE
Calculate new Dgun projectile dir from weaponMuzzlePos

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
instead of from aimpoint. This is how all other weapons determine new projectile dirs. 

A simple change to solve this issue
https://github.com/beyond-all-reason/spring/issues/695
And this issue
https://github.com/beyond-all-reason/Beyond-All-Reason/issues/1900

### CODE REVIEW

File: rts/Sim/Weapons/DGunWeapon.cpp [hunk 1]
```diff
@@ -12,7 +12,8 @@ CR_REG_METADATA(CDGunWeapon, )
 
 void CDGunWeapon::FireImpl(const bool scriptCall)
 {
-	float3 dir = wantedDir;
+	float3 dir = currentTargetPos - weaponMuzzlePos;
+	const float dist = dir.LengthNormalize();
```
Comment (lhog) [inline rts/Sim/Weapons/DGunWeapon.cpp:16]: `dist` looks unused here

Comment (KyleAnthonyShepherd) [reply rts/Sim/Weapons/DGunWeapon.cpp]: Good catch. I had just copied over from LaserCannon.cpp, where it does use dist later. 
But DGunWeapon.cpp does not need that value.

**Comment by sprunk** [comment]
What happens if you `emit-sfx` it via script? Right now I think it just uses the piece orientation, will the "current target pos" be valid if you're just shooting without a target?

**Comment by KyleAnthonyShepherd** [comment]
I can test, but it should have the same behavior as a `emit-sfx` LaserCannon

**Comment by KyleAnthonyShepherd** [comment]
> What happens if you `emit-sfx` it via script? Right now I think it just uses the piece orientation, will the "current target pos" be valid if you're just shooting without a target?

https://github.com/beyond-all-reason/spring/blob/BAR105/rts/Sim/Units/Scripts/UnitScript.cpp#L581
Just tested `emit-sfx` and it seems to behave as expected. `emit-sfx` sets the current target `currentTargetPos` and `weaponMuzzlePos` based on the piece it is emitting from.

**Review by sprunk — APPROVED** [review]

**Pull request merged**